slimt/CMakeLists.txt

```diff
@@ -116,7 +115,7 @@ if(EXPORT_CMAKE_FILE)
 
   # This will double install, but is-ok.
   install(
-    TARGETS slimt-shared slimt-static
+    TARGETS slimt-shared slimt-static sentencepiece
```

Comment:
Is it work if we don't sentencepiece ? (if we have it install in system)

Reply:
This entire block will only work if it is system sentencepiece (i.e installed via [libsentencepiece-dev](https://packages.debian.org/stable/libdevel/libsentencepiece-dev) in debian based machines, [sentencepiece-git](https://aur.archlinux.org/packages/sentencepiece-git) maybe from ArchLinux AUR). For distributions where this is not available, the recommendation is to independently (of slimt) install sentencepiece from sources. If it's sentencepiece from sources, the code will fail at configure step. 

`sentencepiece` only provides the `pkg-config` based [`sentencepiece.pc.in`](https://github.com/google/sentencepiece/blob/8cbdf13794284c30877936f91c6f31e2c1d5aef7/sentencepiece.pc.in). This is used in our builds to induct the 
`sentencpiece` target (a.k.a `Sentencepiece::Sentencepiece`). This happens [here](https://github.com/jerinphilip/slimt/blob/a972cd6e52ba7b762b0614200193bb2d67acce34/3rd-party/CMakeLists.txt#L17-L39) in source.

What we are doing here is merely exposing this target as `slimt::sentencepiece`, as a dependency for `slimt::slimt-shared`. i.e, it refers to the already installed sentencepiece from elsewhere. This target inside cmake's packaging is not provided by the original source, and since we are providing it as `slimt::sentencepiece` it should not be an issue.

You can look at the source of the generated `${CMAKE_INSTALL_PREFIX}/lib/cmake/slimt/slimtTargets.cmake` to see how this is laid out in a real installation. On my installation, it looks like [this](https://gist.github.com/jerinphilip/1f95cb3f4ff9f8f3d2709f821c344807).  

I'm no expert, just thinking out loud here. I'm experimenting with CMakeLists.txt and figuring out where these things go.